firmware/application/file.hpp

```diff
@@ -248,6 +248,7 @@ bool is_directory(const file_status s);
 bool is_regular_file(const file_status s);
 bool file_exists(const path& file_path);
 bool is_directory(const path& file_path);
+bool is_nonempty_directory(const path& file_path);
```

Comment:
nit: would `is_empty_directory` be easier? Would make something like `!is_nonempty_directory` clearer.